### PR TITLE
Add export functionality to reports endpoint

### PR DESCRIPTION
**Pull request by sliverc**
Additional endpoint supporting exporting of reports into csv, ods and xlsx.

### CODE REVIEW

File: timed/tracking/tests/test_report.py [hunk 1]
```diff
@@ -242,3 +259,41 @@ def test_absence_update_on_create_report(self):
             Absence.objects.get(pk=absence.pk).duration ==
             employment.worktime_per_day - timedelta(hours=1)
         )
+
+
+class TestReportHypo(TestCase):
```
Comment (czosel) [inline timed/tracking/tests/test_report.py]: :+1: for using hypothesis

File: timed/tracking/tests/test_report.py [hunk 2]
```diff
@@ -242,3 +259,41 @@ def test_absence_update_on_create_report(self):
...
+                min_delta=timedelta(0),
+                max_delta=timedelta(days=1)
+            )
+        )
+    )
+    @settings(timeout=5)
+    def test_report_export(self, file_type, report):
+        User.objects.create_user(username='test', password='1234qwer')
+        client = JSONAPIClient()
+        client.login('test', '1234qwer')
```
Comment (czosel) [inline timed/tracking/tests/test_report.py]: Are you creating a new user just to make sure that you see reports of other users?

File: timed/tracking/views.py [hunk 3]
```diff
@@ -75,6 +78,38 @@ class ReportViewSet(ModelViewSet):
...
+                report.task.project.customer.name,
+                report.task.project.name,
+                report.task.name,
+                report.user.username,
+                report.comment,
+            ]
+            for report in queryset
+        ]
+
+        file_type = request.query_params.get('file_type')
```
Comment (czosel) [inline timed/tracking/views.py]: Is it really fine to not check that `file_type` is in `['csv', 'xlsx', 'ods']`?

**Comment by czosel** [comment]
And another nitpick: I always prefer to write commit messages in present tense, it makes the history much more readable (e.g. "add export functionality")

File: timed/tracking/tests/test_report.py (hunk 2 above)
Comment (sliverc) [inline timed/tracking/tests/test_report.py]: Hypothesis doesn't support setUp methods properly hence I moved it into its own class which also needs to be subclassed of a special hypothesis TestCase to support transaction per example.

Hence it is needed to create a user here to be able to login.

Comment (czosel) [inline timed/tracking/tests/test_report.py]: Alright, thanks!

File: timed/tracking/views.py (hunk 3 above)
Comment (sliverc) [inline timed/tracking/views.py]: I agree. Adjusted.

**Comment by sliverc** [comment]
Also added blacklist of Cs category in hypothesis test.

Read for review again.

**Comment by czosel** [comment]
Awesome, LGTM :+1:

**Comment by sliverc** [comment]
Merging then.

**Pull request merged**